.travis.yml

```diff
@@ -18,87 +15,56 @@ env:
     - COVERALLS_ENABLED="false"
 
 matrix:
-
   fast_finish: true
-
   include:
-
-    - php: 7.1
     - php: 7.1
       env:
-        - PHPUNIT_ENABLED="false"
         - PHPCSFIXER_ENABLED="true"
-    - php: 7.1
-      env:
-        - SYMFONY_VERSION=4.0.*
-        - COMPOSER_FLAGS="--prefer-lowest"
-    - php: 7.2
-    - php: 7.2
-      env: SYMFONY_VERSION=3.4.*
-    - php: 7.2
-      env: SYMFONY_VERSION=4.0.*
-    - php: 7.2
-      env:
-        - DEPENDENCIES="symfony/phpunit-bridge:^4"
-        - COMPOSER_UPDATE_FLAGS="--no-dev"
     - php: 7.2
+    - php: 7.3
       env:
         - COVERALLS_ENABLED="true"
         - PHPUNIT_FLAGS="-v --coverage-text --coverage-clover var/build/clover.xml"
-    - php: 7.2
+
+    # Minimum supported dependencies with the latest and oldest supported PHP versions
+    - php: 7.1
       env:
-        - SYMFONY_VERSION=dev-master
-        - STABILITY=dev
+        - COMPOSER_FLAGS="--prefer-stable --prefer-lowest"
+    - php: 7.3
+      env:
+        - COMPOSER_FLAGS="--prefer-stable --prefer-lowest"
+
+    # Test nightly PHP with dev dependencies
     - php: nightly
       env:
-        - SYMFONY_VERSION=4.0.*
+        - STABILITY=dev
         - COMPOSER_FLAGS="--ignore-platform-reqs"
-
   allow_failures:
-
-    - env:
-        - COVERALLS_ENABLED="true"
-        - PHPUNIT_FLAGS="-v --coverage-text --coverage-clover var/build/clover.xml"
-    - env:
-        - SYMFONY_VERSION=dev-master
-        - STABILITY=dev
-    - php: nightly
-
+      - env: STABILITY=dev
+      - php: nightly
 
 before_install:
-
-  - if [[ "$SYMFONY_VERSION" != "" ]]; then
-      travis_retry composer require "symfony/symfony:${SYMFONY_VERSION}" --no-update $COMPOSER_FLAGS;
-    fi
-  - if [[ "$DEPENDENCIES" != "" ]]; then
-      travis_retry composer require $DEPENDENCIES --no-update $COMPOSER_FLAGS;
-    fi
   - if [[ "$STABILITY" != "" ]]; then
       travis_retry composer config minimum-stability $STABILITY;
     fi
   - if [[ "$COVERALLS_ENABLED" != "true" ]]; then
       phpenv config-rm xdebug.ini || true;
     fi
   - if [[ "$COVERALLS_ENABLED" == "true" ]]; then
-      travis_retry composer require --dev satooshi/php-coveralls:^2.0@dev --no-update $COMPOSER_FLAGS;
+      travis_retry composer require --dev satooshi/php-coveralls --no-update $COMPOSER_FLAGS;
```

Comment:
what if version 3 of php-coveralls introduces a new set of rules? this would randomly start to fail once they release, no?

Reply:
I will just remove the `@dev` then.